hoot-core/src/main/cpp/hoot/core/io/HootApiDbReader.cpp

```diff
@@ -74,40 +74,17 @@ void HootApiDbReader::open(QString urlStr)
   initializePartial();
 
   QUrl url(_url);
-  QStringList pList = url.path().split("/");
   LOG_DEBUG("Opening database for reader at: " << url.path() << "...");
-  bool ok;
   _database->open(url);
 
-  long requestedMapId = pList[pList.size() - 1].toLong(&ok);
-
-  if (!ok)
+  LOG_VARD(_email);
+  if (!_email.isEmpty())
   {
-    if (_email == "")
-    {
-      throw HootException("If a map name is specified then the user email must also be specified "
-                          "via: " + ConfigOptions::getApiDbEmailKey());
-    }
-
-    QString mapName = pList[pList.size() - 1];
     _database->setUserId(_database->getUserId(_email, false));
-    set<long> mapIds = _database->selectMapIds(mapName);
-    if (mapIds.size() != 1)
-    {
-      QString str =
-        QString("Expected 1 map with the name '%1' but found %2 maps.")
-          .arg(mapName)
-          .arg(mapIds.size());
-      throw HootException(str);
-    }
-    requestedMapId = *mapIds.begin();
   }
 
-  if (!_database->mapExists(requestedMapId))
-  {
-    _database->close();
-    throw HootException("No map exists with ID: " + QString::number(requestedMapId));
-  }
+  const long requestedMapId = _database->getMapIdFromUrl(url);
```

Comment:
There's two queries inside that method.  The first one uses the current user and the second doesn't use a user at all.  It still wasn't quite right, though, since the query with no user should only be bringing back public maps.  The db reading code had too much logic that was shared with the db writing code and made these changes really difficult.  I spent most of today separating that logic where it made sense.